src/Atmos/Parameterizations/CloudPhysics/Microphysics.jl

```diff
@@ -1,272 +1,807 @@
+    _Δv::FT = Δv(snow_param_set)
+
+    return (
+        _n0,
+        _r0,
+        _m0,
+        _me,
+        _χm,
+        _Δm,
+        _a0,
+        _ae,
+        _χa,
+        _Δa,
+        _v0,
+        _ve,
+        _χv,
+        _Δv,
+    )
+end
 
-where:
-  - `q_rai` - rain water specific humidity
-  - `ρ`     - density of air
 
-Returns the mass weighted average rain terminal velocity assuming
-Marshall Palmer 1948 distribution of rain drops.
 """
-function terminal_velocity(param_set::APS, q_rai::FT, ρ::FT) where {FT <: Real}
+    lambda(q, ρ, n0, m0, me, r0, χm, Δm)
+
+ - `q` - specific humidity of rain, ice or snow
+ - `ρ` - air density
+ - `n0` - size distribution parameter
+ - `m0`, `me`, `χm`, `Δm`, `r0` - mass(radius) parameters
 
-    rain_w = FT(0)
+Returns the rate parameter of the assumed size distribution of
+particles (rain drops, ice crystals, snow crystals).
+"""
+function lambda(
+    q::FT,
+    ρ::FT,
+    n0::FT,
+    m0::FT,
+    me::FT,
+    r0::FT,
+    χm::FT,
+    Δm::FT,
+) where {FT <: Real}
 
-    if q_rai > FT(0)
-        v_c = terminal_velocity_single_drop_coeff(param_set, ρ)
-        # gamma(9/2)
-        gamma_9_2 = FT(11.631728396567448)
-        _grav::FT = grav(param_set)
-        _ρ_cloud_liq::FT = ρ_cloud_liq(param_set)
-        _MP_n_0::FT = MP_n_0(param_set)
+    λ::FT = FT(0)
 
-        lambda::FT = (FT(8) * π * _ρ_cloud_liq * _MP_n_0 / ρ / q_rai)^FT(1 / 4)
-        rain_w = gamma_9_2 * v_c / FT(6) * sqrt(_grav / lambda)
+    if q > FT(0)
+        λ =
+            (
+                χm * m0 * n0 * gamma(me + Δm + FT(1)) / ρ / q / r0^(me + Δm)
+            )^FT(1 / (me + Δm + 1))
     end
+    return λ
+end
+
+"""
+    τ_relax(liquid_param_set)
+    τ_relax(ice_param_set)
 
-    return rain_w
+ - `liquid_param_set` - abstract set with cloud liquid water parameters
+ - `ice_param_set` - abstract set with cloud ice parameters
+
+Returns the relaxation timescale for condensation and evaporation of
+cloud liquid water or the relaxation timescale for sublimation and
+deposition of cloud ice.
+"""
+function τ_relax(liquid_param_set::ALPS)
+
+    _τ_relax = τ_cond_evap(liquid_param_set)
+    return _τ_relax
+end
+function τ_relax(ice_param_set::AIPS)
+
+    _τ_relax = τ_sub_dep(ice_param_set)
+    return _τ_relax
 end
 
 """
-    conv_q_vap_to_q_liq(q_sat, q)
+    supersaturation(param_set, q, ρ, T, Liquid())
+    supersaturation(param_set, q, ρ, T, Ice())
 
-where:
-- `q_sat` - PhasePartition at equilibrium
-- `q`     - current PhasePartition
+ - `param_set` - abstract set with earth parameters
+ - `q` - phase partition
+ - `ρ` - air density,
+ - `T` - air temperature
+ - `Liquid()`, `Ice()` - liquid or ice phase to dispatch over.
 
-Returns the q_liq tendency due to condensation/evaporation.
-The tendency is obtained assuming a relaxation to equilibrium with
-constant timescale.
+Returns supersaturation (qv/qv_sat -1) over water or ice.
 """
-function conv_q_vap_to_q_liq(
+function supersaturation(
```

Comment:
Not necessarily in this PR but, should this be moved to Thermodynamics.jl?

Reply:
Just to clarify. What I need here is supersaturation over either water or ice. For example the `relative_humidity()` from `Thermodynamics.jl` returns a weighted sum of relative humidities over water and ice. So the philosophy is different. 

I could add it next to the `q_vap_saturation_generic()` function though.   Would that work?

Reply:
@charleskawczynski - I moved it to Thermodynamics.jl I did not add it to API docs because it's meant to be an internal function.